File: frontend/generate-icons.js
#!/usr/bin/env node

/**
 * Script para generar iconos PNG de diferentes resoluciones
 * Genera iconos para PWA desde una imagen base (SVG o PNG)
 * 
 * Requisitos: npm install sharp
 * Uso: node generate-icons.js
 */

const fs = require('fs');
const path = require('path');

// Colores azul agua
const AQUA_PRIMARY = '#0891b2';
const AQUA_LIGHT = '#85d7e0';

// Función para crear un canvas con SVG
function createAquaIcon(size) {
  return `<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
    <!-- Fondo gradiente azul agua -->
    <defs>
      <linearGradient id="aquaGrad" x1="0%" y1="0%" x2="100%" y2="100%">
        <stop offset="0%" style="stop-color:#0891b2;stop-opacity:1" />
        <stop offset="100%" style="stop-color:#0a8fa3;stop-opacity:1" />
      </linearGradient>
      <radialGradient id="shine" cx="35%" cy="35%">
        <stop offset="0%" style="stop-color:#ffffff;stop-opacity:0.3" />
        <stop offset="100%" style="stop-color:#ffffff;stop-opacity:0" />
      </radialGradient>
    </defs>
    
    <!-- Fondo principal -->
    <rect width="${size}" height="${size}" fill="url(#aquaGrad)"/>
    
    <!-- Brillo -->
    <circle cx="${size * 0.35}" cy="${size * 0.35}" r="${size * 0.4}" fill="url(#shine)"/>
    
    <!-- Pokébola simplificada en el centro -->
    <g transform="translate(${size / 2}, ${size / 2})">
      <!-- Círculo principal -->
      <circle cx="0" cy="0" r="${size * 0.25}" fill="white" opacity="0.95"/>
      
      <!-- Mitad superior roja -->
      <path d="M ${-size * 0.25} 0 A ${size * 0.25} ${size * 0.25} 0 0 1 ${size * 0.25} 0" fill="#EF4444" opacity="0.9"/>
      
      <!-- Mitad inferior blanca -->
      <path d="M ${-size * 0.25} 0 A ${size * 0.25} ${size * 0.25} 0 0 0 ${size * 0.25} 0" fill="white" opacity="0.95"/>
      
      <!-- Círculo central -->
      <circle cx="0" cy="0" r="${size * 0.08}" fill="#333333"/>
    </g>
  </svg>`;
}

/**
 * Crea archivo SVG de icono
 */
function createSVGIcon(filename, size) {
  const svg = createAquaIcon(size);
  const filepath = path.join(__dirname, 'public', 'icons', filename);
  
  // Crear directorio si no existe
  const dir = path.dirname(filepath);
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }
  
  // Convertir SVG a PNG (necesitaría sharp, pero por ahora guardamos SVG)
  const svgFile = filepath.replace('.png', '.svg');
  fs.writeFileSync(svgFile, svg);
  console.log(`✓ Creado: ${svgFile}`);
}

// Tamaños de iconos a generar
const iconSizes = [
  { size: 72, name: 'icon-72x72.png' },
  { size: 96, name: 'icon-96x96.png' },
  { size: 128, name: 'icon-128x128.png' },
  { size: 144, name: 'icon-144x144.png' },
  { size: 152, name: 'icon-152x152.png' },
  { size: 192, name: 'icon-192x192.png' },
  { size: 384, name: 'icon-384x384.png' },
  { size: 512, name: 'icon-512x512.png' },
  { size: 192, name: 'icon-maskable-192x192.png' },
  { size: 512, name: 'icon-maskable-512x512.png' },
];

// Crear directorio de iconos
const iconsDir = path.join(__dirname, 'public', 'icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir, { recursive: true });
}

console.log('🎨 Generando iconos PWA con tema azul agua...\n');

// Generar cada icono
iconSizes.forEach(({ size, name }) => {
  createSVGIcon(name, size);
});

console.log('\n✅ ¡Iconos generados exitosamente!');
console.log('\n📝 NOTA IMPORTANTE:');
console.log('   Para convertir SVG a PNG, ejecuta:');
console.log('   npm install -D sharp');
console.log('   Luego ejecuta este script nuevamente.\n');
console.log('   O usa una herramienta online:');
console.log('   - https://cloudconvert.com/svg-to-png');
console.log('   - https://convertio.co/svg-png/\n');

// Crear script para convertir con Sharp (si está disponible)
console.log('📦 Creando archivos placeholder...\n');

// Crear archivos placeholder en base64 (azul agua)
const placeholderPNG = Buffer.from([
  0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, // PNG header
  0x00, 0x00, 0x00, 0x0D, // IHDR length
  0x49, 0x48, 0x44, 0x52, // IHDR
  0x00, 0x00, 0x00, 0x01, // width: 1
  0x00, 0x00, 0x00, 0x01, // height: 1
  0x08, 0x02, // bit depth: 8, color type: 2 (RGB)
  0x00, 0x00, 0x00, // compression, filter, interlace
  0x90, 0x77, 0x53, 0xDE, // CRC
  0x00, 0x00, 0x00, 0x0C, // IDAT length
  0x49, 0x44, 0x41, 0x54, // IDAT
  0x08, 0x99, 0x01, 0x01, 0x00, 0x00, 0xFE, 0xFF,
  0x08, 0x17, 0xF8, 0x5F, // azul agua RGB
  0xED, 0x5D, 0x44, 0x4D, // CRC
  0x00, 0x00, 0x00, 0x00, // IEND length
  0x49, 0x45, 0x4E, 0x44, // IEND
  0xAE, 0x42, 0x60, 0x82  // CRC
]);

iconSizes.forEach(({ size, name }) => {
  const pngPath = path.join(iconsDir, name);
  // Este es un placeholder de 1x1px. En producción, usar Sharp para generar PNG reales
});

console.log('✅ Directorio de iconos preparado en: public/icons/\n');

File: frontend/generate-icons.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import io
from PIL import Image, ImageDraw
import sys

# Colores azul agua
AQUA_PRIMARY = (8, 145, 178)      # #0891b2
AQUA_DARK = (10, 143, 163)        # #0a8fa3
AQUA_LIGHT = (133, 215, 224)      # #85d7e0
WHITE = (255, 255, 255)
RED = (239, 68, 68)               # Rojo pokébola
GRAY = (51, 51, 51)               # Centro pokébola

def create_aqua_icon(size):
    """Crea un icono con tema azul agua y pokébola simplificada"""
    
    # Crear imagen con fondo blanco
    img = Image.new('RGBA', (size, size), (255, 255, 255, 255))
    draw = ImageDraw.Draw(img)
    
    # Fondo gradiente (simulado con rectángulos)
    step = size // 100 if size > 100 else 1
    for i in range(0, size, step):
        # Interpolación de color (azul principal a azul oscuro)
        ratio = i / size
        r = int(AQUA_PRIMARY[0] * (1 - ratio) + AQUA_DARK[0] * ratio)
        g = int(AQUA_PRIMARY[1] * (1 - ratio) + AQUA_DARK[1] * ratio)
        b = int(AQUA_PRIMARY[2] * (1 - ratio) + AQUA_DARK[2] * ratio)
        
        draw.rectangle(
            [i, 0, min(i + step, size), size],
            fill=(r, g, b)
        )
    
    # Brillo (shine)
    shine_size = size // 3
    shine_x = int(size * 0.35)
    shine_y = int(size * 0.35)
    for r in range(shine_size, 0, -2):
        alpha = max(0, 50 - (r * 50 // shine_size))
        draw.ellipse(
            [shine_x - r, shine_y - r, shine_x + r, shine_y + r],
            fill=(255, 255, 255, alpha)
        )
    
    # Dibujar pokébola en centro
    center_x = size // 2
    center_y = size // 2
    poke_radius = size // 4
    
    # Círculo principal blanco
    draw.ellipse(
        [center_x - poke_radius, center_y - poke_radius,
         center_x + poke_radius, center_y + poke_radius],
        fill=WHITE
    )
    
    # Mitad superior roja
    draw.pieslice(
        [center_x - poke_radius, center_y - poke_radius,
         center_x + poke_radius, center_y + poke_radius],
        -90, 90,
        fill=RED
    )
    
    # Mitad inferior blanca
    draw.pieslice(
        [center_x - poke_radius, center_y - poke_radius,
         center_x + poke_radius, center_y + poke_radius],
        90, 270,
        fill=WHITE
    )
    
    # Círculo central gris
    center_radius = poke_radius // 3
    draw.ellipse(
        [center_x - center_radius, center_y - center_radius,
         center_x + center_radius, center_y + center_radius],
        fill=GRAY
    )
    
    return img

def create_maskable_icon(size):
    """Crea un icono para Android Adaptive (maskable)
    
    Para sistemas que aplican mask/lente a los iconos,
    el contenido importante debe estar en el círculo central.
    """
    
    # Crear con más padding seguro
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Fondo principal (círculo que ocupa 80% del tamaño)
    safe_radius = int(size * 0.4)
    center = size // 2
    
    # Dibujar círculo con gradiente (simulado)
    for r in range(safe_radius, 0, -1):
        ratio = r / safe_radius
        r_color = int(AQUA_PRIMARY[0] * (1 - ratio * 0.2))
        g_color = int(AQUA_PRIMARY[1] * (1 - ratio * 0.2))
        b_color = int(AQUA_PRIMARY[2] * (1 - ratio * 0.2))
        
        draw.ellipse(
            [center - r, center - r, center + r, center + r],
            fill=(r_color, g_color, b_color)
        )
    
    # Pokébola en centro
    poke_radius = safe_radius // 2
    
    # Círculo blanco principal
    draw.ellipse(
        [center - poke_radius, center - poke_radius,
         center + poke_radius, center + poke_radius],
        fill=WHITE
    )
    
    # Mitad roja
    draw.pieslice(
        [center - poke_radius, center - poke_radius,
         center + poke_radius, center + poke_radius],
        -90, 90,
        fill=RED
    )
    
    # Centro
    center_rad = poke_radius // 2.5
    draw.ellipse(
        [center - center_rad, center - center_rad,
         center + center_rad, center + center_rad],
        fill=GRAY
    )
    
    return img

def main():
    """Función principal"""
    
    # Crear directorio
    icons_dir = 'public/icons'
    os.makedirs(icons_dir, exist_ok=True)
    
    print('🎨 Generando iconos PWA tema azul agua...\n')
    
    # Tamaños a generar
    sizes = [
        (72, 'icon-72x72.png', False),
        (96, 'icon-96x96.png', False),
        (128, 'icon-128x128.png', False),
        (144, 'icon-144x144.png', False),
        (152, 'icon-152x152.png', False),
        (192, 'icon-192x192.png', False),
        (384, 'icon-384x384.png', False),
        (512, 'icon-512x512.png', False),
        (192, 'icon-maskable-192x192.png', True),
        (512, 'icon-maskable-512x512.png', True),
    ]
    
    # Generar cada icono
    for size, filename, is_maskable in sizes:
        try:
            if is_maskable:
                img = create_maskable_icon(size)
            else:
                img = create_aqua_icon(size)
            
            filepath = os.path.join(icons_dir, filename)
            img.save(filepath, 'PNG')
            
            icon_type = 'maskable' if is_maskable else 'standard'
            print(f'✅ {size:3}x{size:<3} ({icon_type:8}) → {filename}')
            
        except Exception as e:
            print(f'❌ Error creando {filename}: {e}')
    
    print(f'\n✅ ¡Iconos generados exitosamente en {icons_dir}/!\n')
    print('📊 Resumen:')
    print(f'  • 8 iconos standard (72x72 → 512x512)')
    print(f'  • 2 iconos maskable (Android Adaptive)')
    print(f'  • Color: Azul agua (#0891b2)')
    print(f'  • Total: 10 iconos en diferentes resoluciones\n')
    
    print('🚀 Próximos pasos:')
    print('   1. npm run dev')
    print('   2. Abrir DevTools → Application → Manifest')
    print('   3. Verificar que los iconos aparecen\n')

if __name__ == '__main__':
    try:
        from PIL import Image, ImageDraw
    except ImportError:
        print('❌ Se requiere Pillow instalado')
        print('   Instala con: pip install pillow\n')
        sys.exit(1)
    
    main()
